internal/worker/json.go

```diff
@@ -21,10 +21,22 @@ type OSBuildJob struct {
 }
 
 type OSBuildJobResult struct {
-	Success       bool            `json:"success"`
-	OSBuildOutput *osbuild.Result `json:"osbuild_output,omitempty"`
-	TargetErrors  []string        `json:"target_errors,omitempty"`
-	UploadStatus  string          `json:"upload_status"`
+	Success       bool                   `json:"success"`
+	OSBuildOutput *osbuild.Result        `json:"osbuild_output,omitempty"`
+	TargetResults []*target.TargetResult `json:"target_results,omitempty"`
+	TargetErrors  []string               `json:"target_errors,omitempty"`
+	UploadStatus  string                 `json:"upload_status"`
+}
+
+type TargetResult struct {
+	Type string      `json:"type"` // aws, gcp, azure, ....
+	Data interface{} `json:"data"` //cast this to a specific result depending on the type
+}
+
+/* if targetResult.Type == "aws" { targetResult.Data.(TargetAWSResult).Ami } */
+type TargetAWSResult struct {
```

Comment:
I'm a bit confused, but it seems that the `TargetResult` and `TargetAWSResult` defined here are actually not used anywhere. Instead, for internal communication with worker, the structure from `target` package is used and for cloudapi the one generated by `openapi`.

Reply:
Indeed, I think these can just be removed. I think it's just noise from my initial faffing about https://github.com/Gundersanne/osbuild-composer/commit/2e0475b061f996de9995814952b6cf1b649e59bd